src/components/sticker.rs

```diff
@@ -0,0 +1,191 @@
+use adw::prelude::*;
+use adw::subclass::prelude::*;
+use glib::clone;
+use gtk::{gio, glib};
+
+use tdlib::enums::StickerFormat;
+use tdlib::types::Sticker as TdSticker;
+
+use crate::session::Session;
+use crate::utils::{decode_image_from_path, spawn};
+
+mod imp {
+    use super::*;
+    use std::cell::{Cell, RefCell};
+
+    #[derive(Default)]
+    pub struct Sticker {
+        pub(super) message_id: Cell<i64>,
```

Comment:
Am I wrong or this property seems completely useless? Also, it doesn't make sense to have it anyway because if I understand correctly we're gonna use this widget outside of the messages as well (like for a sticker picker).

Reply:
I want to check if widget isn't recycled after sticker is downloaded IIRC you have a similar check in MessageDocument

Reply:
Ah k, yeah that check is important. The problem is that I don't like the fact that it's placed in the generic component. Maybe it's better having a sync variant to load the sticker and still do the check in MessageSticker?